dask/array/tests/test_ghost.py

```diff
@@ -160,3 +163,93 @@ def test_ghost():
 
     g = ghost(d, depth={0: 2, 1: 1}, boundary={0: 100})
     assert g.chunks == ((8, 8), (5, 5))
+
+
+def test_map_overlap():
+    x = da.arange(10, chunks=5)
+
+    y = x.map_overlap(lambda x: x + len(x), depth=2)
+    assert eq(y, np.arange(10) + 5 + 2 + 2)
+
+
+def test_nearest_ghost():
+    a = np.arange(144).reshape(12, 12).astype(float)
+
+    darr = da.from_array(a, chunks=(6, 6))
+    garr = ghost(darr, depth={0: 5, 1: 5},
+                 boundary={0: 'nearest', 1: 'nearest'})
+    tarr = trim_internal(garr, {0: 5, 1: 5})
+    assert_array_almost_equal(np.array(tarr), a)
```

Comment:
One of the nice things about the current `eq` is that it computes with a one of the synchronous schedulers like `dask.get` or `dask.async.get_sync`, both of which can be traversed with a debugger if things go wrong.  Otherwise arrays compute by default with the async threaded scheduler